core/dbt/task/test.py

```diff
@@ -79,37 +92,52 @@ def execute_test(
         table = result['table']
         num_rows = len(table.rows)
         if num_rows != 1:
-            num_cols = len(table.columns)
-            # since we just wrapped our query in `select count(*)`, we are in
-            # big trouble!
             raise InternalException(
                 f"dbt internally failed to execute {test.unique_id}: "
-                f"Returned {num_rows} rows and {num_cols} cols, but expected "
-                f"1 row and 1 column"
+                f"Returned {num_rows} rows, but expected "
+                f"1 row"
+            )
+        num_cols = len(table.columns)
+        if num_cols != 5:
+            raise InternalException(
+                f"dbt internally failed to execute {test.unique_id}: "
+                f"Returned {num_cols} columns, but expected "
+                f"5 columns"
             )
-        return int(table[0][0])
+
+        test_result_data: PrimitiveDict = dict(zip(table.column_names, table.rows[0]))
```

Comment:
Seeing this in the Snowflake test:
```
Unhandled error while executing test.test.unique_view_model_id.d4e7c60bfe
Field "validation_errors" of type builtins.int is missing in dbt.task.test.TestResult instance
```
I'm guessing this is because Snowflake returns all uppercase identifiers, so the key is `VALIDATION_ERRORS`. I'm sure there's a way to handle that